vllm/config.py

```diff
@@ -300,6 +303,17 @@ def _verify_cuda_graph(self) -> None:
         self.max_seq_len_to_capture = min(self.max_seq_len_to_capture,
                                           self.max_model_len)
 
+    def _verify_output_proc_callback(self) -> None:
+        logger.warning(
+                    "To see benefits of output processor callback, enable CUDA "
+                    "graph. Since, enforce-eager is enabled, output processor callback "
+                    "cannot be used")
+        self.use_output_proc_callback = not self.enforce_eager
+        # TO DO: assert no beam search
```

Comment:
beam search asserts cannot be checked at engine initialization. Since beam search params are known only when a request's sampling params `best_of` and `n` are set at runtime. So can throw error message at the beginning of `step`